### ISSUE TITLE
`brew tap-new` fails if user's gitconfig has `commit.gpgsign true` and the GPG key requires a pinentry

### ISSUE THREAD

**Issue by apetresc** [Homebrew/brew#10992]
### `brew config` output

```shell
HOMEBREW_VERSION: 3.0.10-80-g9fb771d
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9fb771d2a4d9d6a3b0677abf6915b06a2767cddf
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 4e9060dd297d223a2066e02ea0f163b6bbbb4426
Core tap last commit: 44 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DEVELOPER: set
HOMEBREW_EDITOR: nvim
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.31.1 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4
Rosetta 2: false
```

### `brew doctor` output

```shell
Your system is ready to brew.
```

### 

The `brew doctor` above contains no "Warning" lines.

### What were you trying to do (and why)?

I was trying to use `brew tap-new` to create a new tap. My git config happens to set `commit.gpgsign true`, and my private key is stored on a Yubikey smartcard (which was plugged in).

(I should add that manually committing from the command line works just fine)

### What happened (include all command output)?

```
$ brew tap-new --branch master apetresc/homebrew-magic
Initialized empty Git repository in /opt/homebrew/Library/Taps/apetresc/homebrew-magic/.git/
error: cannot run gpg: No such file or directory
error: gpg failed to sign the data
fatal: failed to write commit object
Error: Failure while executing; `git commit -m Create\ apetresc/magic\ tap` exited with 128.
```

Basically, the `brew tap-new` command failed halfway through. It complains that the problem is that it can't find `gpg` but I suspect the real root cause is that it's unable to run the pinentry program for some reason.

### What did you expect to happen?

I expect the same thing that happens if I run the failing command (`git commit -m Create\ apetresc/magic\ tap`) directly from the exact same CLI session: my pinentry program comes up, I enter the smartcard PIN, and it signs the commit for me.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1. Configure your git according to best practices, with `commit.gpgsign true` and private keys stored on a hardware device such as a Yubikey smartcard, with a PIN.
2. Attempt to run `brew tap-new`.
```

**Comment by nandahkrishna** [comment]
Do you have `gpg.program` set in your `git` configuration?

## `gpg.program` is not set
```
➜ git config --global --get gpg.program
➜ brew tap-new --branch master nandahkrishna/testing
Initialized empty Git repository in /usr/local/Homebrew/Library/Taps/nandahkrishna/homebrew-testing/.git/
error: cannot run gpg: No such file or directory
error: gpg failed to sign the data
fatal: failed to write commit object
Error: Failure while executing; `git commit -m Create\ nandahkrishna/testing\ tap` exited with 128.
```

However, committing works fine:
```
➜ touch README.md
➜ git add README.md
➜ git commit -m "Test"
[master (root-commit) edd67c9] Test
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 README.md
```

## `gpg.program` is set
```
➜ git config --global gpg.program $(which gpg)
➜ git config --global --get gpg.program
/usr/local/bin/gpg
➜ brew tap-new --branch master nandahkrishna/testing
Initialized empty Git repository in /usr/local/Homebrew/Library/Taps/nandahkrishna/homebrew-testing/.git/
[master (root-commit) eb8ac9d] Create nandahkrishna/testing tap
 3 files changed, 88 insertions(+)
 create mode 100644 .github/workflows/publish.yml
 create mode 100644 .github/workflows/tests.yml
 create mode 100644 README.md
==> Created nandahkrishna/testing
/usr/local/Homebrew/Library/Taps/nandahkrishna/homebrew-testing

When a pull request making changes to a formula (or formulae) becomes green
(all checks passed), then you can publish the built bottles.
To do so, label your PR as `pr-pull` and the workflow will be triggered.
```

**Comment by apetresc** [comment]
Ah, @nandahkrishna, you're right, explicitly setting `gpg.program` does "fix" the problem.

I'm guessing this is because `brew` operates in an environment with no `$PATH` set, for security reasons? Is there any way around this? (The reason I don't usually set `gpg.program` is because it makes my `.gitconfig` not easily syncable between machines. And normal `git` will find `gpg` on `$PATH` under normal circumstances anyway)

**Comment by nandahkrishna** [comment]
The `tap-new` command should probably ensure that `gpg` is available in the `PATH` (using `Utils::Git.setup_gpg!`), just as in `brew pr-pull` and some other commands. I briefly tested this and it seemed to work.

I'm not sure if this would cause any other issues, but here's a patch, if you'd be able to create a PR:
```diff
diff --git a/Library/Homebrew/dev-cmd/tap-new.rb b/Library/Homebrew/dev-cmd/tap-new.rb
index fee8572a3..b977cb0b2 100644
--- a/Library/Homebrew/dev-cmd/tap-new.rb
+++ b/Library/Homebrew/dev-cmd/tap-new.rb
@@ -149,6 +149,9 @@ module Homebrew
 
     unless args.no_git?
       cd tap.path do
+        Utils::Git.set_name_email!
+        Utils::Git.setup_gpg!
+
         # Would be nice to use --initial-branch here but it's not available in
         # older versions of Git that we support.
         safe_system "git", "-c", "init.defaultBranch=#{branch}", "init"
```

**Comment by apetresc** [comment]
Thank you so much, @nandahkrishna!

**Comment by nandahkrishna** [comment]
No problem! Thanks for reporting this.
